<!-- -->
<launch>
  <arg name="ns" default="trajectory_follower"/>
  <arg name="debug" default="false"/>

  <arg name="uses_lerp" default="True"/>
  <arg name="publish_twist_cmd" default="False"/>
  <arg name="publish_ctrl_cmd" default="True"/>
  <arg name="reverse_minimum_lookahead_distance" default="7.0"/>
  <arg name="control_period" default="0.033"/>
  <arg name="velocity_delay_compensation_time" default="0.57"/>
  <arg name="use_lateral_error_compensation" default="false"/>
  <arg name="lateral_error_compensation_ratio" default="1.0"/>
  <arg name="lateral_error_compensation_max" default="2.0"/>
  <arg name="steering_delay_compensation_time" default="0.5"/>
  <arg name="use_steering_delay_compensation" default="false"/>

  <group ns="$(arg ns)">
    <!-- rosrun waypoint_follower pure_pursuit -->
    <node pkg="pure_pursuit" type="pure_pursuit_rev2" name="pure_pursuit" output="log">
      <param name="uses_lerp" value="$(arg uses_lerp)"/>
      <param name="publish_twist_cmd" value="$(arg publish_twist_cmd)"/>
      <param name="publish_ctrl_cmd" value="$(arg publish_ctrl_cmd)"/>
      <param name="reverse_minimum_lookahead_distance" value="$(arg reverse_minimum_lookahead_distance)"/>
      <param name="control_period" value="$(arg control_period)"/>
      <param name="velocity_delay_compensation_time" value="$(arg velocity_delay_compensation_time)"/>
      <param name="use_lateral_error_compensation" value="$(arg use_lateral_error_compensation)"/>
      <param name="lateral_error_compensation_ratio" value="$(arg lateral_error_compensation_ratio)"/>
      <param name="lateral_error_compensation_max" value="$(arg lateral_error_compensation_max)"/>
      <param name="steering_delay_compensation_time" value="$(arg steering_delay_compensation_time)"/>
      <param name="use_steering_delay_compensation" value="$(arg use_steering_delay_compensation)"/>
    </node>

    <group if="$(arg debug)">
      <!-- rosrun rosservice rosservice call /pure_pursuit/set_logger_level ros.pure_pursuit DEBUG -->
      <node pkg="rosservice" type="rosservice" name="pure_pursuit_set_logger_level_debug"
          args="call pure_pursuit/set_logger_level ros.pure_pursuit DEBUG"/>
    </group>
  </group>

</launch>
